### DISCUSSION TITLE
Symfony 6 - get parameter from .env and use it outside of Controller

### DISCUSSION THREAD

**Discussion by petr-ekl-itcube** [symfony/symfony#49092] [Q&A]
I need help with subject:
I use Commands in my app, as it is command line based;
I use dependency injection in constructor here but it is allready enough of dependencies here.
Command instantiates some component X,Y,Z etc... that do something and this is manualy instantiating them as I build the name of the class to do particular work, so dependency injection here is not possible
 To be concrete Command  receives what to do and chooses which Componet will do it

Components X,Y,Z uses ClinetAPIs. ClientApi is class which needs parameter API_URL from .env and it is more ClientAPIs than one

I wonder how to get API_URL_X from .env to concrete ClientAPI_X

Injection in Command not make sense, it's too much possible classes could need to inject

I don't know how to instantiate service ClientAPI_X leeting it to inject param from .env by services.xml, suppose it's not possible

I don't know hoe to get anyhow param from .env in ClientAPi_X....

I've read anything about autowiringand injection with no clue, any help, please?

**Comment by MatTheCat** [comment]
> it's too much possible classes could need to inject

That’s what [services locators](https://symfony.com/doc/current/service_container/service_subscribers_locators.html) are for.

**Reply by petr-ekl-itcube** [reply]
Thanks, this seems as a solution. Little bit complicated but hope thet right one.

And with just getting parameter from .env anywhere? At least as hotfix until I make service locator work. Nothing?

**Reply by MatTheCat** [reply]
I didn't understand why you want to get parameters from “anywhere” 🤔

Can't you register your services using the `%env()%` syntax?

**Reply by petr-ekl-itcube** [reply]
Thank you very much. It's much more easy than it is described.
It would be GREAT to put link to this approach anywhere in manual around autowiring and services description. I've spent around month of extensive searching until I've put my question and received your answer.
